### ISSUE TITLE
Pupil Player Crashes Error

### ISSUE THREAD

**Issue by emolyzr** [pupil-labs/pupil#1166]
[player.log](https://github.com/pupil-labs/pupil/files/1957885/player.log)
<img width="576" alt="unbenannt" src="https://user-images.githubusercontent.com/35568338/39398343-9541a146-4b0c-11e8-87a0-bc1c0f56f183.png">

How can this error be solved (see log file) ?

**Comment by emolyzr** [comment]
EDIT: same issue with another data set!
Need to know whether the data sets are corrupted or if it's a software or hardware issue (DELL Latitude E5450, WIN 10 Pro, Intel i5 5300U und 8GB RAM).

[player.log](https://github.com/pupil-labs/pupil/files/1957913/player.log)
<img width="576" alt="unbenannt" src="https://user-images.githubusercontent.com/35568338/39398719-f85ec8ee-4b11-11e8-828a-0deb565dfa16.png">

**Comment by papr** [comment]
Please update Pupil Player to version v1.6

**Comment by emolyzr** [comment]
thanks for the fast reply. already did:
pupil_player_windows_x64_v1.6.11

**Comment by papr** [comment]
Please delete the user settings if you still see the issue after updating. These are the files that end with `_user_settings` in the `pupil_player_settings` folder in your home directory.

**Comment by emolyzr** [comment]
Interesting bug. Issue solved. Many thanks @papr !

**Comment by emolyzr** [comment]
Pupil Player v.1.6.11 crashes, while exporting video
video export did not start yet.
attachet the play.log file
and the export_infos.csv which was exported

Can you help please?
I try again with deleting the user_setting files.



PLAYER.LOG


2018-04-30 11:24:20,565 - MainProcess - [INFO] os_utils: Disabling idle sleep not supported on this OS version.
2018-04-30 11:24:22,772 - player - [ERROR] player_methods: No valid dir supplied
2018-04-30 11:24:22,777 - player - [INFO] launchables.player: Session setting are from a  different version of this app. I will not use those.
2018-04-30 11:24:29,074 - player - [INFO] launchables.player: Starting new session with 'C:\Users\admin\Desktop\P1_DATA\TV_SAT1_P1'
2018-04-30 11:24:29,079 - player - [INFO] player_methods: Updating meta info
2018-04-30 11:24:29,082 - player - [INFO] player_methods: Checking for world-less recording
2018-04-30 11:24:29,774 - player - [INFO] video_capture: Install pyrealsense to use the Intel RealSense backend
2018-04-30 11:24:30,468 - player - [INFO] launchables.player: Application Version: 1.6.11
2018-04-30 11:24:30,469 - player - [INFO] launchables.player: System Info: User: admin, Platform: Windows, Machine: DESKTOP-S260J1A, Release: 10, Version: 10.0.16299
2018-04-30 11:24:30,565 - player - [INFO] camera_models: Previously recorded calibration found and loaded!
2018-04-30 11:24:30,566 - player - [INFO] launchables.player: Session setting are a different version of this app. I will not use those.
2018-04-30 11:37:09,401 - player - [INFO] camera_models: No user calibration found for camera eye0 at resolution (192, 192)
2018-04-30 11:37:09,403 - player - [INFO] camera_models: No pre-recorded calibration available
2018-04-30 11:37:09,404 - player - [WARNING] camera_models: Loading dummy calibration
2018-04-30 11:37:09,455 - player - [INFO] camera_models: No user calibration found for camera eye1 at resolution (192, 192)
2018-04-30 11:37:09,455 - player - [INFO] camera_models: No pre-recorded calibration available
2018-04-30 11:37:09,456 - player - [WARNING] camera_models: Loading dummy calibration
2018-04-30 11:42:05,387 - player - [INFO] launchables.player: Created export dir at "C:\Users\admin\Desktop\P1_DATA\TV_SAT1_P1\exports\000"
2018-04-30 11:42:05,538 - player - [WARNING] video_export_launcher: Adding new video export process.
2018-04-30 11:44:58,652 - player - [ERROR] launchables.player: Process Player crashed with trace:
Traceback (most recent call last):
  File "launchables\player.py", line 457, in player
  File "shared_modules\video_export_launcher.py", line 96, in on_notify
  File "shared_modules\video_export_launcher.py", line 118, in add_export
  File "shared_modules\background_helper.py", line 37, in __init__
  File "multiprocessing\process.py", line 105, in start
  File "multiprocessing\context.py", line 223, in _Popen
  File "multiprocessing\context.py", line 322, in _Popen
  File "multiprocessing\popen_spawn_win32.py", line 65, in __init__
  File "multiprocessing\reduction.py", line 60, in dump
OSError: [Errno 22] Invalid argument

2018-04-30 11:44:58,680 - player - [INFO] launchables.player: Process shutting down.


EXPORT_INFO.CSV


key,value
--
 
Player Software Version,1.6.11
 
Data Format Version,1.6.11
 
Export Date,30.04.2018
 
Export Time,11:42:05
 
Frame Index Range:,0 - 40899
 
Relative Time Range,00:00.000 - 28:08.564
 
Absolute Time Range,3299.208772 -   4987.773448

**Comment by papr** [comment]
```
2018-04-30 11:44:58,652 - player - [ERROR] launchables.player: Process Player crashed with trace:
Traceback (most recent call last):
File "launchables\player.py", line 457, in player
File "shared_modules\video_export_launcher.py", line 96, in on_notify
File "shared_modules\video_export_launcher.py", line 118, in add_export
File "shared_modules\background_helper.py", line 37, in init
File "multiprocessing\process.py", line 105, in start
File "multiprocessing\context.py", line 223, in _Popen
File "multiprocessing\context.py", line 322, in _Popen
File "multiprocessing\popen_spawn_win32.py", line 65, in init
File "multiprocessing\reduction.py", line 60, in dump
OSError: [Errno 22] Invalid argument
```

This is a multiprocessing issue that I am not able to reproduce on our Windows testing machine.

**Comment by emolyzr** [comment]
Thanks @papr for your response. Does this mean the best solution would be to use another Windows machine? (serious question) Or can you imagine another solution? Thanks, for any hint.

**Comment by papr** [comment]
I do not know what the cause for it could be. Trying a different system would be a good idea.

**Comment by emolyzr** [comment]
*Again, a new error*. Any help would be highly appreciated. Attached via wetransfer are two data sets, that lead to the error. Streamed over WIFI Network from a Google Nexus 5p (006) and Motorola Z2 Play (003) during a trial setup to figure out how far the network connection reaches.

https://we.tl/TQmEpsBMLJ

PLAYER LOG for 003 (identical with 006):
2018-05-06 16:19:54,312 - MainProcess - [INFO] os_utils: Disabling idle sleep not supported on this OS version.
2018-05-06 16:19:54,754 - player - [ERROR] player_methods: No valid dir supplied
2018-05-06 16:19:58,077 - player - [INFO] launchables.player: Starting new session with 'C:\Users\emolyzr1\recordings\2018_05_06\003'
2018-05-06 16:19:58,087 - player - [INFO] player_methods: Updating meta info
2018-05-06 16:19:58,091 - player - [INFO] player_methods: Checking for world-less recording
2018-05-06 16:19:58,525 - player - [INFO] video_capture: Install pyrealsense to use the Intel RealSense backend
2018-05-06 16:19:58,761 - player - [INFO] launchables.player: Application Version: 1.6.11
2018-05-06 16:19:58,761 - player - [INFO] launchables.player: System Info: User: emolyzr1, Platform: Windows, Machine: DESKTOP-KPBA42L, Release: 10, Version: 10.0.16299
**2018-05-06 16:19:58,769 - player - [ERROR] video_capture.file_backend: Videofile pts are not evenly spaced, pts to index conversion may fail and be inconsitent.**
2018-05-06 16:19:58,782 - player - [INFO] camera_models: Previously recorded calibration found and loaded!
2018-05-06 16:20:01,948 - player - [WARNING] seek_control: End of video - restart at beginning.
2018-05-06 16:20:14,792 - player - [WARNING] seek_control: End of video - restart at beginning.
2018-05-06 16:20:16,296 - player - [INFO] launchables.player: Process shutting down.

**Comment by papr** [comment]
My guess what happened is that time sync on the phone changed the frames' timestamps while you were recording in Capture. There is now way for the phone to know if Capture is recording. We recommend to make recordings on the phone itself since the phone will not change timestamps during a local recording. Streaming to capture is meant for monitoring, not as a save way to record the video.

**Comment by emolyzr** [comment]
This means, together with the comment by moritz to another issue (https://github.com/pupil-labs/pupil/issues/1167#issuecomment-385979120), that currently there is no safe way to record data with the pupil mobile app while streaming, am I correct?

Direct recording to the phone without streaming is the only safe way? Is it safe? Or are there any know bugs?

Do you think the files that I attached to my last comment are repairable or is the data defenitely lost?

**Comment by papr** [comment]
See my comment on the other issue.

Regarding to the streamed recording in Pupil Capture. I just realized that this is not an issue with time sync but how Pupil Capture saves the video file. I will try to replicate and fix the issue this week.

**Comment by emolyzr** [comment]
hi @papr, these were just test recordings. But it would be bad, if this happens in a real study and data would be lost. Thanks for your engagement!

**Comment by mkassner** [comment]
fixed with version 1.8!
